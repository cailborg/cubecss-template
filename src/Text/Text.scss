.button {
  color: white;
  background-color: white;
  & svg {
    fill: white;
    margin-left: var(--lp-space-xxxs);
  };
  &:focus {
    outline:none;
  }
  &:active {
    position: relative;
    &::before {
      content: "";
      position: absolute;
      z-index: -1;
      width: 100%;
      height: 100%;
      background: black;
      border-radius: var(--lp-radii-sm);
    }
  }
  &.large {
    height: var(--lp-space-xxl);
    font-size: var(--lp-font-size-button-lg);
    font-weight: var(--lp-font-weight-button-lg);
    line-height: var(--lp-line-height-button-lg);
    letter-spacing: var(--lp-letter-spacing-button-lg);
    padding: 0 var(--lp-space-md);
  }
  &.medium {
    height: var(--lp-space-lg);
    font-size: var(--lp-font-size-button-md);
    font-weight: var(--lp-font-weight-button-md);
    line-height: var(--lp-line-height-button-md);
    letter-spacing: var(--lp-letter-spacing-button-md);
    padding: 0 var(--lp-space-xs);
    min-width: 160px;
  }
  &.small {
    height: var(--lp-space-md);
    font-size: var(--lp-font-size-button-sm);
    font-weight: var(--lp-font-weight-button-sm);
    line-height: var(--lp-line-height-button-sm);
    letter-spacing: var(--lp-letter-spacing-button-sm);
    padding: 0 var(--lp-space-xs);
    min-width: 96px;
  }
  &[data-variant="primary"] {
    background: rgba(var(--lp-colors-primary), 1);
    &:hover, &:active {
      background: rgba(var(--lp-colors-primary), 0.8);
    }
  }
  &[data-variant="reversed"] {
    background: rgba(var(--lp-colors-neutral-50), 1);
    color: rgba(var(--lp-colors-primary), 1);
    & svg {
      fill: rgba(var(--lp-colors-primary), 1);
    }
    &:hover {
      background: white;
    }
    &:active {
      background: rgba(var(--lp-colors-neutral-100), 1);
    }
  }
  &[data-variant="destructive"] {
    background: rgba(var(--lp-colors-persimmon-600), 1);
    color: rgba(var(--lp-colors-neutral-900), 1);
    & svg {
      fill: rgba(var(--lp-colors-neutral-900), 1);
    }
    &:hover {
      background: rgba(var(--lp-colors-persimmon-500), 1);
    }
    &:active {
      background: rgba(var(--lp-colors-persimmon-700), 1);
    }
  }
  &[data-variant="ghost"] {
    background: none;
    color: rgba(var(--lp-colors-primary), 1);
    & svg {
      fill: rgba(var(--lp-colors-primary), 1);
    }
    &:hover, &:active {
      color: rgba(var(--lp-colors-primary), 0.8);
      & svg {
        margin-left: var(--lp-space-xxs);
      }
    }
    &:active {
      &::before {
        background: none;
      }
    }
  }
  &:disabled {
    background: rgba(var(--lp-colors-neutral-50), 1);
    color: rgba(var(--lp-colors-neutral-400), 1);
    & svg {
      fill: rgba(var(--lp-colors-neutral-400), 1);
    };
  }
  &.focus-ring {
    box-shadow: 0 0 0 2px white, 0 0 0 6px dodgerblue;
  }
}
