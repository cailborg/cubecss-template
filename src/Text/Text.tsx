import React from "react";
import styled from "@emotion/styled";

export interface TextProps {
    variant?: "body-1" | "body-2" | "body-3" | "caption" | "overline";
    children: React.ReactNode;
    tagName?: "p" | "span";
    alignment?: "left" | "right" | "center";
    id?: string;
    isEmphasised?: boolean;
    className?: string;
}

export const Text: React.FC<TextProps> = ({
    variant,
    children,
    tagName = "p",
    alignment,
    id,
    className,
    isEmphasised = false,
}) => {
    const TextTag = styled.span`
        text-align: ${alignment ?? "inherit"};
        font-family: var(--lp-fonts-${variant});
        font-size: var(--lp-font-size-${variant});
        line-height: var(--lp-line-height-${variant});
        letter-spacing: var(--lp-letter-spacing-${variant});
        text-transform: var(--lp-text-transform-${variant});
        + p {
            margin-top: var(--lp-paragraph-spacing-${variant});
        }
        font-weight: ${isEmphasised ? "700" : "400"};
    `;
    return (
        <TextTag className={className} as={`${tagName}`} id={id}>
            {children}
        </TextTag>
    );
};
