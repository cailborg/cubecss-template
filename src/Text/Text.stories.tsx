import { blue } from "@material-ui/core/colors";
import { Story, Meta } from "@storybook/react/types-6-0";
import { Text, TextProps } from "./Text";

export default {
  title: "Components/Text",
  component: Text,
  argTypes: {
    children: { control: "text" },
  },
} as Meta;

export const Sentence: Story<TextProps> = (args) => <Text {...args}>{args.children}</Text>;
Sentence.args = { children: 'Example Text', variant:"body1", tagName:"p", alignment: "left" };

export const Paragraph: Story<TextProps> = (args) => <div><Text {...args}>{args.children}</Text> <Text {...args}>{args.children}</Text></div>;
Paragraph.args = { children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras sit amet urna sed leo laoreet scelerisque. Pellentesque condimentum arcu lacinia lacus pulvinar scelerisque. Proin interdum, erat auctor iaculis blandit, mi tellus sollicitudin dui, nec imperdiet mauris lorem non nisl. Etiam libero ex, posuere at maximus quis, sagittis eleifend felis. Nam non leo est. Suspendisse potenti. Suspendisse feugiat, mauris sed iaculis tincidunt, risus ex blandit nibh, quis bibendum ligula tortor ac nibh. Mauris vel fringilla felis. Donec quis vestibulum lacus.", variant:"body3", tagName:"p", alignment: "left" };