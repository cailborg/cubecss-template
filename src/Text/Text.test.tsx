import React from "react";
import { Text } from "./Text";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Text>Hello</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});
