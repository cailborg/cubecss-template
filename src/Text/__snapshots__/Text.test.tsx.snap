// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className="css-ymnrku-Text"
>
  Hello
</p>
`;
